src/js/hooks/productSupplier/form/useProductSupplierForm.js

```diff
@@ -111,6 +121,19 @@ const useProductSupplierForm = () => {
     console.log(payload);
   };
 
+  const packagePrice = useWatch({ control, name: 'packagePrice' });
+  const packageSize = useWatch({ control, name: 'packageSize' });
+
+  // eachPrice is a computed value from packagePrice and packageSize
+  useEffect(() => {
+    if (packagePrice && packageSize) {
+      setValue('eachPrice', decimalParser(packagePrice / packageSize, 4));
+    } else {
+      setValue('eachPrice', '');
+    }
+  },
+  [packagePrice, packageSize]);
```

Comment:
If you decide to create the `usePackageSpecification` hook, I think this could be placed there, but you would have to return the `setValue` from this hook and pass it to the `PricingSection` from `ProductSupplierFormMain` component.
But on the other hand I don't mind keeping it there - I just think that such calculations that are strictly connected to a specific section, should be handled in this section's hook.

Reply:
I'm 50-50 about this one, so I'll let you guys decide.

Reply:
yes, I see what you mean but I feel like in this case, it is alright to calculate this value in our main hook for product sources. I don't think it adds any complexity to the readability of the code since it is the same place where we transform/manipulate the form data for the API call, but that's my opinion.